### PR TITLE
[WEBSITE-959] Allow multiple urls in the template document

### PR DESCRIPTION
**Pull request by katylouise**
* Changed implementation to store different template urls and types from the description, replacing the single @base_url.
* Updated documentation.
* Fixed some of the rubocop violations.

### CODE REVIEW

File: lib/parliament/request/open_search_request.rb [hunk 1]
```diff
@@ -71,32 +75,41 @@ def get_description(url)
...
 
       def query_url
         @query_url
       end
 
-      def setup_query_url(search_params)
+      def setup_query_url(search_params, type = nil)
         search_terms = search_params[:query]
-        query_url = @base_url.dup
+        type ||= 'application/atom+xml'
```
Comment (mattrayner) [inline lib/parliament/request/open_search_request.rb:108]: should this not just be `type = 'application/atom+xml'` in the parameters above?

**Review by mattrayner — APPROVED** [review]

**Review by mattrayner — CHANGES_REQUESTED** [review]
see above

File: lib/parliament/request/open_search_request.rb [hunk 2]
```diff
@@ -71,32 +75,41 @@ def get_description(url)
...
+
+        url_hash = @templates[:url].select { |url| url[:type] == type }.first
```
Comment (mattrayner) [inline lib/parliament/request/open_search_request.rb:110]: What if this is nil?

**Review by mattrayner — CHANGES_REQUESTED** [review]
Have we tested the scenario where there is no description URL with a provided type in the description file?

I.e. I want type application/json but there isn't one available

File: lib/parliament/request/open_search_request.rb (hunk 2 above)
Comment (katylouise) [reply lib/parliament/request/open_search_request.rb:110]: good point, I'll add testing around this

File: lib/parliament/builder/open_search_response_builder.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ module Builder
     #
     # @since 0.1.0
     class OpenSearchResponseBuilder < Parliament::Builder::BaseResponseBuilder
-      OPEN_SEARCH_ELEMENTS = %w(totalResults Query startIndex itemsPerPage).freeze
+      OPEN_SEARCH_ELEMENTS = %w[totalResults Query startIndex itemsPerPage].freeze
```
[the remark this answers is not in the document]
Comment (mattrayner) [reply lib/parliament/builder/open_search_response_builder.rb]: should these be changed back?

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -10,16 +10,25 @@
...
+
+# Setup the initial description file request
+
 VCR.configure do |config|
   config.cassette_library_dir = 'spec/fixtures/vcr_cassettes'
   config.hook_into :webmock # or :fakeweb
   config.configure_rspec_metadata!
+
+  # Dynamically filter our sensitive information
+  config.filter_sensitive_data('<AUTH_TOKEN>') { ENV['OPENSEARCH_AUTH_TOKEN'] }
```
Comment (mattrayner) [inline spec/spec_helper.rb:27]: We should also filter the API url

Comment (mattrayner) [reply spec/spec_helper.rb:27]: See prototype for implementation

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -10,16 +10,25 @@
...
 end
 
 RSpec.configure do |config|
+
```
Comment (mattrayner) [inline spec/spec_helper.rb]: needed?

**Review by mattrayner — CHANGES_REQUESTED** [review]

File: spec/spec_helper.rb (hunk 4 above)
Comment (katylouise) [reply spec/spec_helper.rb:27]: done

**Review by mattrayner — APPROVED** [review]

**Pull request merged**